new/property.md

```diff
@@ -0,0 +1,223 @@
+---
+RFC: unassigned
+Author: John Keiser <jkeiser@chef.io>
+Status: Draft
+Type: Standards Track
+---
+
+# Resource Attribute Improvements
+
+We add a number of enhancements to `attribute`:
+
+- Make nil a valid value (`path nil`)
+- Make a nicer type / validation syntax (`property :path, String`)
+- Add lazy defaults.
+- Add coercion.
+- Add `property` (an alias to `attribute`) to `Chef::Resource` to make it available to all users.
+- Add `property_is_set?(:property_name)`
+
+## Motivation
+
+    As a Chef user,
+    I want to be able to use natural syntax for properties,
+    So that I can spend less time writing cookbooks, and make them more readable.
+
+    As a Chef user,
+    I want resources to be more readable and ubiquitous,
+    So that I can easily tell the interface to things I use.
+
+    As a Chef user,
+    I want resource attributes and node attributes to use different words,
+    so that they don't lead me to conflate them as concepts.
+
+## Specification
+
+### Make nil a valid value
+
+It will now be possible to set a property to `nil` by saying `my_property nil`.  (Currently, this will not change the value of `my_property`.)
+
+In Chef 12, we will keep behavior the same, and *deprecate* setting a property to `nil`.  We will allow properties to explicitly allow `nil`, however, by specifying it explicitly as a valid value: `property :path, [ String, nil ]`.
+
+### Make defaults lazy
+
+`lazy` defaults are automatically run in the context of the *instance*:
+
+```ruby
+class MysqlInstall < Chef::Resource
+  property :root_path, String, default: '/'
+  property :config_path, String, default: lazy { File.join(root_path, 'config') }
+end
+```
+
+This is a breaking change, and in Chef 12 will only affect properties (not attributes).
```

Comment:
So this might be a breaking change if the user did `attribute :y, default: Chef::DelayedEvaluator.new { foo }`, but I'm comfortable saying no one has ever done that :-)